File: C++/Move_Zeroes.cpp
/* Given an integer array nums, move all 0's to the end of it while maintaining the relative order of the non-zero elements */


class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int zero = 0;
        int i = 0;
        
        while (i < nums.size() - zero) {
            if (nums[i] == 0) {
                zero++;
                nums.push_back(0);
                nums.erase(nums.begin() + i);
            } else {
                i++;
            }      
        }
    }
};

File: Swift/Move_Zeroes.swift
/* Given an integer array nums, move all 0's to the end of it while maintaining the relative order of the non-zero elements */

class Solution {
    func moveZeroes(_ nums: inout [Int]) {
        var i = 0
        var counter = 0
        while i < nums.count - counter {
            if nums[i] == 0 {
                counter += 1
                nums.append(0)
                nums.remove(at: i)
            } else {
                i += 1
            }
        }
    }
}